generators/client/templates/angular/src/main/webapp/app/shared/login/_login.service.ts

```diff
@@ -91,7 +91,10 @@ export class LoginService {
     <%_ } _%>
 
     logout() {
-        this.authServerProvider.logout().subscribe();
-        this.principal.authenticate(null);
+        if (this.principal.isAuthenticated()) {
```

Comment:
I try this code and it works for both (default app and uaa app):
```typescript
    logout() {
        if (this.principal.isAuthenticated()) {
            this.authServerProvider.logout().subscribe();
        }
        this.principal.authenticate(null);
    }
```

What do you think?

Reply:
This wasn't working for UAA because it would call `authenticate` before the logout completed and it would return account information.